probe-rs/src/rtt/channel.rs

```diff
@@ -295,10 +295,10 @@ impl Channel {
                 .filter_map(|mr| mr.as_ram_region())
                 .merge_consecutive()
                 .any(|rr| {
-                    rr.range.contains(&self.info.buffer_start_pointer())
-                        && rr.range.contains(
-                            &(self.info.buffer_start_pointer() + self.info.size_of_buffer()),
-                        )
+                    let start = self.info.buffer_start_pointer();
+                    let end = self.info.buffer_start_pointer() + self.info.size_of_buffer();
+
+                    rr.range.contains(&start) && end <= rr.range.end
```

Comment:
From what I understand, the issue was that `self.info.buffer_start_pointer() + self.info.size_of_buffer()` points to the address *after* the end of the buffer, so that the `contains` check doesn't work, right?

Please add a comment explaining why `contains` is only used for one end here.